sklearn/preprocessing/_data.py

```diff
@@ -2748,8 +2749,30 @@ def _dense_fit(self, X, random_state):
             X = resample(
                 X, replace=False, n_samples=self.subsample, random_state=random_state
             )
+            if sample_weight is not None:
+                sample_weight = sample_weight[: X.shape[0]]
+
+        self.quantiles_ = np.zeros((len(references), n_features))
+
+        for i in range(n_features):
+            col = X[:, i]
+            mask = ~np.isnan(col)
+            col_clean = col[mask]
```

Comment:
`_weighted_percentile` has builtin support to ignore nan values, so this should no longer be needed.